src/components/implementation/no_interface/new_boot/boot_deps.h

```diff
@@ -0,0 +1,147 @@
+#include <print.h>
+
+#undef assert
+#define assert(node) do { if (unlikely(!(node))) { debug_print("assert error in @ "); *((int *)0) = 0;} } while(0)
+
+#include <cos_alloc.h>
+#include <cobj_format.h>
+#include <cos_types.h>
+
+/*the booter uses this to keep track of each comp*/
+struct comp_cap_info {
+	struct cos_compinfo cos_compinfo;
```

Comment:
Put it in another way, do you have other pointers referencing to the same `cos_compinfo`?
If so, there should be a pointer.